aptos-move/framework/aptos-framework/sources/account_recovery.move

```diff
@@ -0,0 +1,148 @@
+        // sign for transactions in the `mint_event_ticket()` function.
+        move_to(signer, ModuleData {
+            signer_cap: resource_signer_cap,
+        });
+    }
+
+    fun recovery_exists(addr: address): bool {
+        exists<AccountRecovery>(addr)
+    }
+
+    public fun register(account: &signer,
+                        authorized_addresses: vector<address>,
+                        required_delay_seconds: u64,
+                        rotation_capability_sig_bytes: vector<u8>,
+                        account_public_key_bytes: vector<u8>,
+    ) acquires AccountRecovery, AccountRecoveryReverseLookup {
+        let addr = signer::address_of(account);
+
+        assert!(!recovery_exists(addr), ERECOVERY_ALREADY_SET);
+        assert!(!exists<AccountRecoveryReverseLookup>(@source_addr));
+
+        let reverse_lookup = borrow_global_mut<AccountRecoveryReverseLookup>(@source_addr);
+
+        move_to(account, AccountRecovery {
+            authorized_addresses,
+            required_delay_seconds,
+            account_recovery_init: option::none<>(),
+        });
+
+        let i = 0;
+        let len = vector::length(&authorized_addresses);
+        while (i < len) {
+            let authorized_address = *vector::borrow(&authorized_addresses, i);
+            let list = table::borrow_mut_with_default(&mut reverse_lookup.authorized_to_recovery, authorized_address, vector::empty<address>());
+            vector::push_back(list, addr);
+        };
+        account::offer_rotation_capability(signer, rotation_capability_sig_bytes, 0, account_public_key_bytes, @source_addr)
+    }
+
+    public fun initiate_account_key_recovery(account: &signer, recovery_address: address) acquires AccountRecovery, AddressToAccountRecovery {
+        assert!(recovery_exists(recovery_address), ERECOVERY_NOT_SET);
+
+        let account_recovery = borrow_global_mut<AccountRecovery>(recovery_address);
+        let addr = signer::address_of(account);
+        assert!(vector::contains(&account_recovery.authorized_addresses, &addr));
+        assert!(account_recovery.account_recovery_init);
+
+        account_recovery.account_recovery_init = std::option::some(AccountRecoveryInitData{
+            recovery_seq_number: account::get_sequence_number(recovery_address),
+            recovery_initiation_ts: timestamp::now_seconds(),
+        });
+    }
+
+    public fun rotate_key(delegated_signer: &signer,
+                          recovery_address: address,
+                          new_public_key_bytes: vector<u8>,
+                          cap_update_table: vector<u8>) acquires AccountRecovery, ModuleData {
+        assert!(recovery_exists(recovery_address), ERECOVERY_NOT_SET);
+
+        let account_recovery = borrow_global_mut<AccountRecovery>(recovery_address);
+        let delegated_addr = signer::address_of(delegated_signer);
+        assert!(vector::contains(&account_recovery.authorized_addresses, &delegated_addr));
```

Comment:
one account can only delegate their rotation capability to one other account. what's the vector `authorized_addresses` for?